dev/Scroller/Scroller.cpp

```diff
@@ -5650,6 +5673,8 @@ void Scroller::ChangeOffsetsPrivate(
     winrt::SnapPointsMode snapPointsMode = options ? options.SnapPointsMode() : ScrollOptions::s_defaultSnapPointsMode;
     InteractionTrackerAsyncOperationType operationType;
 
+    animationMode = GetComputedAnimationMode(animationMode);
```

Comment:
Is it possible to combine the definition of animationMode and the redefinition using getComputedAnimationMode() into one helper?

Reply:
I would need two new helpers, for ScrollOptions and ZoomOptions.  I'm leaving it as is.

Reply:
I don't think I get it. seems like every use of animationMode looks like:

    winrt::AnimationMode animationMode = options ? options.AnimationMode() : ScrollOptions::s_defaultAnimationMode;


    animationMode = GetComputedAnimationMode(animationMode);

that bit of code is repeated a lot.

---
In reply to: [293043515](https://github.com/microsoft/microsoft-ui-xaml/pull/850#discussion_r293043515) [](ancestors = 293043515)